rubicon_ml/viz/experiments_table.py

```diff
@@ -0,0 +1,281 @@
+        self.experiment_records = []
+        self.github_url = None
+        self.metric_names = set()
+        self.parameter_names = set()
+        self.all_columns = ["id", "name", "created_at", "model_name", "commit_hash", "tags"]
+        self.hidden_columns = []
+
+        commit_hashes = set()
+        show_columns = {"id", "created_at"}
+
+        for experiment in self.experiments:
+            if experiment.commit_hash is not None:
+                commit_hashes.add(experiment.commit_hash)
+                show_columns.add("commit_hash")
+
+            if experiment.model_name is not None:
+                show_columns.add("model_name")
+
+            if experiment.name is not None:
+                show_columns.add("name")
+
+            if len(experiment.tags) > 0:
+                show_columns.append("tags")
+
+            experiment_record = {
+                "id": experiment.id,
+                "name": experiment.name,
+                "created_at": experiment.created_at,
+                "model_name": experiment.model_name,
+                "commit_hash": experiment.commit_hash[:7],
+                "tags": experiment.tags,
+            }
+
+            for parameter in experiment.parameters():
+                experiment_record[parameter.name] = parameter.value
+
+                self.parameter_names.add(parameter.name)
+
+            for metric in experiment.metrics():
+                experiment_record[metric.name] = metric.value
+
+                self.metric_names.add(metric.name)
+
+            self.experiment_records.append(experiment_record)
+
+        self.metric_names = list(self.metric_names)
+        self.parameter_names = list(self.parameter_names)
+
+        self.all_columns.extend(self.parameter_names + self.metric_names)
+        self.hidden_columns = [
+            column
+            for column in self.all_columns
+            if column not in list(show_columns) + self.metric_names + self.parameter_names
+        ]
+
+        if len(commit_hashes) == 1:
+            github_url_root = self.experiments[0].project.github_url[:-4]
+            commit_hash = list(commit_hashes)[0]
+
+            self.commit_hash = commit_hash[:7]
+            self.github_url = f"{github_url_root}/tree/{commit_hash}"
+
+        self.app.layout = self._build_frame(self._build_layout())
+
+        _register_callbacks(self.app)
+
+    def _build_layout(self):
+        bulk_select_buttons = [
+            dbc.Button(
+                "select all experiments",
+                color="primary",
+                disabled=not self.is_selectable,
+                id="select-all-button",
+                outline=True,
+            ),
+            dbc.Button(
+                "clear all experiments",
+                color="primary",
+                disabled=not self.is_selectable,
+                id="clear-all-button",
+                outline=True,
+            ),
+        ]
+
+        experiment_table = dash_table.DataTable(
```

Comment:
When the "show all" button is selected, the "id" column takes up quite a lot of space with empty space on the left side of the ids but the "created_at" column values are blocked. Would it be possible to make the id column smaller and allow for the other columns to be seen more clearly?

Reply:
I'm not seeing the same behavior. maybe it has to do with the size of the browser window you're viewing it in?

Reply:
Looks good now! (Maybe because the buttons were disabled, it fixed itself) Thanks!